themes/classic/templates/_partials/microdata/product-jsonld.tpl

```diff
@@ -0,0 +1,107 @@
+{**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License 3.0 (AFL-3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ *}
+<script type="application/ld+json">
+  {
+    "@context": "https://schema.org/",
+    "@type": "Product",
+    "name": "{$product.name}",
+    "description": "{$page.meta.description|regex_replace:"/[\r\n]/" : " "}",
+    "category": "{$product.category_name}",
+    {if isset($product.cover)}"image" :"{$product.cover.bySize.home_default.url}",{/if}
+    "sku": "{if $product.reference}{$product.reference}{else}{$product.id}{/if}",
+    "mpn": "{if $product.mpn}{$product.mpn}{elseif $product.reference}{$product.reference}{else}{$product.id}{/if}",
```

Comment:
You can add **gtin** here

Reply:
I added it like that in 2709a0f
`{if $product.ean13}"gtin13": "{$product.ean13}",{else if $product.upc}"gtin13": "0{$product.upc}",{/if}
    {if $product.isbn}"isbn": "{$product.isbn}",{/if}`

The Google rich text will always pass because we have a sku and a mnp for sure. gtin and isbn are nice to have when available... (good approach ?)

Reply:
@fdonnet I source information from the official website - https://schema.org/Product I think what would be the cleanest way:
- We can either use gtin13 and prefix UPCs by zero - `{if $product.ean13}"gtin13": "{$product.ean13}",{else if $product.upc}"gtin13": "0{$product.upc}",{/if}` OR we can use the gtin field. They say that you can put all types of GTINs in there.
- Use ISBN like this: `{if $product.isbn}"productID": "{$product.isbn}",{/if}`, They say this: _The product identifier, such as ISBN. For example: meta itemprop="productID" content="isbn:123-456-789"._

Reply:
adapted for upc. For the moment, I removed isbn because maybe the productId field content needs to choosen by the shop owner. We cannot know if we need to put ean13 or isbn etc (and we need to prefix by the choosen type)...
Maybe isbn needs to be stored in gtin13 too..
At the end it is not really important in term of Google warning because it's already covered with sku and mpn fields...
What a mess...

Reply:
@fdonnet I agree. For a proper ISBN display, the item type would need to be a book. https://schema.org/Book#eg-0462 But that needs another properties like author, publisher, bookFormat etc. I don't think this should belong to standard Prestashop core. WDYT?